src/tool/xmeta/xmeta_emit.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+#include "xmeta_models.h"
+#include "xlang_model_listener.h"
+#include <winrt/base.h>
+#include <vector>
+#include <mscoree.h>
+#include <cor.h>
+#include <meta_reader.h>
+#include <meta_writer.h>
+
+namespace xlang::xmeta
+{
+    inline std::string type_semantics_to_string(model_ref<type_semantics> const& semantic_type);
+
+    class xmeta_emit : public xlang_model_listener
+    {
+    public:
+        xmeta_emit(std::string const& assembly_name) 
+            : m_assembly_name(assembly_name)
+        { };
+
+        void initialize();
+        void uninitialize();
+        void save_to_file() const;
+		std::vector<uint8_t> save_to_memory() const;
+
+        void listen_namespace_model(std::shared_ptr<namespace_model> const& model) override;
```

Comment:
Consider `final` rather than `override`.

Reply:
How about both?

Reply:
Saying both is redundant. `final` implies `override`. A function should only have exactly one of `virtual`, `override`, or `final`.

https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rh-override

Reply:
I see. Noted :)